pkgs/servers/nfd/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, stdenv
+, boost
+, fetchFromGitHub
+, libpcap
+, ndn-cxx
+, openssl
+, pkg-config
+, sphinx
+, systemd
+, wafHook
+, websocketpp
+, withSystemd ? stdenv.isLinux
+, withWebSocket ? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nfd";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "named-data";
+    repo = lib.toUpper pname;
+    rev = "NFD-${version}";
+    sha256 = "1l9bchj8c68r6qw4vr1kc96jgxl0vpqa2vjkvy1xmhz92sivr6gi";
+    fetchSubmodules = withWebSocket;
```

Comment:
I don't think this will work the way you're intending - withWebSocket doesn't change the sha256 so the submodules aren't being turned on/off here. I would probably just stick to fetching submodules unconditionally so you don't need complicated logic to change the sha256.

Reply:
You're right, the hash would change. I'll make it fetch the submodules unconditionally.